eliminate support for ancient git and Tk, update to Tcl/Tk 8.6

mark987 opened the pull request:
Based upon j6t/testing to avoid conflicts with inflight topics.

- sets minimum git at 2.20 following prior exchange. Latest advisory touches only back to 2.43, so 2.20 predates anything the project supports. But, updating to anything beyond 1.72 eliminates all version checks and git workarounds, so 2.20 is sufficient for that.
- eliminates non-ttk code paths to simplify code, now compatible with inflight topics, depends upon Tk >= 8.6. Removes many alternate code paths.

The "33 commits" notion above is nonsense: there are 5 commits I'm adding on top of j6t/testing

j6t commented:
Thanks, this is certainly the way to go!

But please do not build on top of j6t-testing. I guess the potential conflict arises from the removal of `${NS}`. In this case, I suggest that we remove all uses that are in master, but do not remove the variable so that it is still available for the topic branches.

mark987 commented:
Removing pre-ttk support depends upon the updates for Tcl/Tk 8.6, basically the last commit in what you carry as 'ml/tcltk-9', a3a7e5ecb686e7bed0014b8469ad3f8919c7639a. There are three major topics convolved:

1. eliminate support for early git
2. updates for Tcl/Tk 8.6
3. updates for Tcl/Tk 8.7+, including 9.0

Unfortunately, these lead to merge conflicts, regardless of order. Any other topic adding / removing a global variable is likely to conflict, so keeping all of the current in-flight topics open and independent is problematic.

I can rearrange ml/tcltk-9 into early git support removal, Tcl 8.6 updates, including ttk removal, then Tcl 8.7/9.0 support if you desire. Or, just let ml/tcltk-9 get integrated to master and I'll return with ttk, etc. later. Let me know.

BTW - I have a similar but larger series for git-gui to get to the same goals (remove early git support, Tcl 8.6, then 8.6/9+). Conflicts abound again there, so I'll follow whatever order we settle on with gitk.

j6t commented:
Would it make sense that you pick the commits from #20 that are necessary for this feature here, and then rebuild the remaining patches of #20 on top of this branch?

mark987 commented:
I've updated this branch to base off of origin/master and provide
* remove support to ancient git
* update requirements to Tcl/Tk 8.6, remove code supporting earlier Tcl/Tk
Nothing here touches Tcl 9.0 support.

mark987 commented:
Thank you for the careful review, I did review and accept all of your comments, added your "Helped-by" on a few where they were not just my rebasing errors (these patches have moved several times). The final commit, 
05dde34, has the same tree as https://github.com/j6t/gitk/tree/ml/abandon-old-version, so there should be no surprises.

I updated PR#20 to branch off of 3e43143 as that topic has no dependency upon the final 3 patches:
- the prior 257aa5e you mentioned taking separately.
- The two patches on the color config dialog are relevant in that they depend upon ttk: the approach does not fully handle quirks of the base widget set in Tk 8.4 so has languished in my tree for years. These two can be deferred as you see fit.

carenas commented:
FWIW, Tcl/TK 8;6 support is missing from Amazon Linux, so this is likely going to prevent anyone using that to be able to use either `git gui` or `gitk` there, unless a third party package is added.

The same is also happening in macOS, which unlike Amazon Linux is more likely to have these tools running regularly, but luckily, third party packages are also more readily available.

j6t commented:
@carenas Is Amazon Linux locked into an old version of Git as well, or does it pick up the current version?

j6t commented:
Any suggestions how to move forward? Or can we declare this somebody else's problem?

mark987 commented:
> *j6t* left a comment (j6t/gitk#22)
> <https://github.com/j6t/gitk/pull/22#issuecomment-3137065728>
>
> Any suggestions how to move forward? Or can we declare this somebody else's problem?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/j6t/gitk/pull/22#issuecomment-3137065728>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABIORPWAGBJFVGZO33UDLQD3LDXZLAVCNFSM6AAAAACBEXFOW2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTCMZXGA3DKNZSHA>.
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

I claim this is someone else's problem, and is a natural consequence of choosing the
particular OS and release. Typically on an LTS platform you get only bug fixes and
security patches for the tools originally shipped, and 10 years is at the upper end of
commonly available support. If a user wants later tools, they need a more up to date
platform; ultimately, you pick Fedora, not Redhat, and live with the instability. I think
accommodating tools and libraries up to 10 years old is a reasonable balance, seems to
comport with git overall, and Tcltk 8.6 is certainly more than 10 years old so 8.5 is
outside that window.
